api/external_auth.proto

```diff
@@ -0,0 +1,151 @@
+syntax = "proto3";
+
+// [#proto-status: draft]
+
+package envoy.api.v2;
+
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+import "google/rpc/status.proto";
+
+// Authorization service.
+// Primarily responds with OK or NOT OK.
+service Authorization {
+  rpc Check(CheckRequest) returns (CheckResponse);
+}
+
+message CheckRequest {
+  // The name of the tenant for multi-tenant support.
+  string tenant_name = 1;
+
+  // The request attributes.
+  AttributeContext attributes = 2;
+}
+
+message CheckResponse {
+  // Status OK allows the request. Any other response indicates the request should be denied.
+  google.rpc.Status status = 1;
+}
+
+// An attribute is a piece of metadata that describes an activity on a network.
+// For example, the size of an HTTP request, or the status code of an HTTP response.
+//
+// Each attribute has a type and a name, which is logically defined as a proto message field.
+// The fields follow the Istio attribute vocabulary.
```

Comment:
I would prefer that attributes here stand alone,  i.e. we don't make Envoy APIs/docs depend on an external document for Istio that is subject to independent versioning or URL movement.